frontend/pages/index.js

```diff
@@ -1,7 +1,65 @@
+import React, { useState } from "react";
 import Head from "next/head";
-import { Section } from "../components";
+import {
+  AddBookModal,
+  Book,
+  Button,
+  EmptyState,
+  Flex,
+  Grid,
+  Layout,
+  Section,
+  Text,
+} from "../components";
+import useToast from "../hooks/useToast";
+import { fontWeight, spacing } from "../theme";
+import { useGetAuthors, useAddAuthor } from "../api/authors";
+import { useGetBooks, useAddBook } from "../api/books";
 
 const Home = () => {
+  const [showAddBookModal, setShowAddBookModal] = useState(false);
+  const { refetchAuthors } = useGetAuthors();
+  const { addAuthor } = useAddAuthor();
+  const { books, booksLoading, booksError } = useGetBooks();
+  const { addBook } = useAddBook();
+  const [_, addToast] = useToast();
+
+  const findAuthor = async (name) => {
+    const authors = await refetchAuthors();
+    const authorIndex = authors
+      ?.map(({ firstName, lastName }) => `${firstName} ${lastName}`)
+      .findIndex((authorName) => authorName === name);
+    if (authorIndex >= 0) {
+      return authors[authorIndex];
+    }
+    return null;
+  };
+
+  const onAddBookModalToggle = () => {
+    setShowAddBookModal(!showAddBookModal);
+  };
+  const onAddBookModalSubmit = async ({
+    title,
+    author: authorName,
```

Comment:
Can you elaborate on why this is necessary?

Reply:
The event handler needs to be asynchronous ("async") because there is a function that returns a promise that needs to be resolved using "await" before the function finishes running.